magento-research/pwa-studio
# enhancement(load): Branded first renders

## Description

Refactor Mustache templates to share a small critical CSS block and use it to render a branded first paint.

Add a general concept of `resource.name` to UPWARD, enabling the loading indicator to be a shared partial which uses `resource.name` and can therefore be customized per page.

Refactor noscript and 404 pages to use the same basic system.

## Related Issue
Closes #1272 

## Verification Steps

1. Go to a nonexistent page with an HTML extension, like `/nothing.html`. **Verify Not Found page.**  
  ![magento-venia-concept-fquwg local pwadev_8436_nothing html(iPhone 5_SE)](https://user-images.githubusercontent.com/1643758/58393075-97685d80-8002-11e9-8822-c9c66696c45d.png)
![magento-venia-concept-fquwg local pwadev_8436_nothing html(iPhone 5_SE) (1)](https://user-images.githubusercontent.com/1643758/58393090-a9e29700-8002-11e9-8f8e-0253e76aaf79.png)

2. Edit `packages/venia-concept/templates/seed-bundles.mst` and type garbage in the `src` attribute so the app doesn't load.

```mustache
{{#assets.bundles.load}}
    <script defer src="bsdasbd{{.}}"></script>
{{/assets.bundles.load}}
```

3. Go to homepage. **Verify branded first render.**

![magento-venia-concept-fquwg local pwadev_8436_(iPhone 5_SE)](https://user-images.githubusercontent.com/1643758/58393146-f29a5000-8002-11e9-8359-b7d821703715.png)
![magento-venia-concept-fquwg local pwadev_8436_(iPhone 5_SE) (1)](https://user-images.githubusercontent.com/1643758/58393159-fe861200-8002-11e9-8ca4-42b779a879e3.png)

4. Go to category page: `/venia-tops.html`. **Verify branded first render.**
![magento-venia-concept-fquwg local pwadev_8436_venia-tops html(iPhone 5_SE)](https://user-images.githubusercontent.com/1643758/58393188-207f9480-8003-11e9-868d-0872711ac28d.png)

5. Go to product page: `/carmina-earrings.html`. **Verify branded first render.**  
  ![magento-venia-concept-fquwg local pwadev_8436_carmina-earrings html(iPhone 5_SE)](https://user-images.githubusercontent.com/1643758/58393193-31300a80-8003-11e9-9141-2d49fb8f2159.png)

6. Open Chrome Developer tools, open the Command Palette (Cmd-Shift-P on OSX) and type "Disable JavaScript". Press Enter.

7. Refresh page. **Verify branded no-script experience.**
![magento-venia-concept-fquwg local pwadev_8436_carmina-earrings html(iPhone 5_SE) (1)](https://user-images.githubusercontent.com/1643758/58393213-5c1a5e80-8003-11e9-944f-ac8100dd3879.png)
![magento-venia-concept-fquwg local pwadev_8436_carmina-earrings html(iPhone 5_SE) (2)](https://user-images.githubusercontent.com/1643758/58393220-650b3000-8003-11e9-8cf4-f64cf4a5b811.png)

## How Have YOU Tested this?
Portrait and landscape orientation on many, many browsers and devices.

## Proposed Labels for Change Type/Package




## Checklist:
